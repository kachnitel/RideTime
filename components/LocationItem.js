import React from 'react';
import { Text, View, StyleSheet } from 'react-native';
import { LocationItemDetail } from './LocationItemDetail';

export class LocationItem extends React.Component {
  render() {
    return <View style={styles.location}>
      <Text style={styles.name}>{this.props.location.name}</Text>
      <LocationItemDetail location={this.props.location} />
      {/* <Text style={styles.itemCoords}>{item.coords[0]}; {item.coords[1]}</Text> */}
    </View>;
  }
}

const styles = StyleSheet.create({
    name: {
      padding: 3,
      fontSize: 16,
      height: 26,
    },
    itemCoords: {
      paddingLeft: 5,
      fontSize: 10,
      height: 12,
    },
    location: {
      height: 64,
      backgroundColor: 'white',
      borderRadius: 4,
      borderWidth: 0.5,
      borderColor: '#d6d7da',
    }
})