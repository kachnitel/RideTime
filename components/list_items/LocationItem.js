import React from 'react'
import { StyleSheet, Text, View } from 'react-native'
import Layout from '../../constants/Layout'
import LocationItemDetail from '../LocationItemDetail'
import PropTypes from 'prop-types'

export default class LocationItem extends React.Component {
  render () {
    return (
      <View style={{ ...styles.listItem, ...this.props.style }}>
        <Text style={{ ...styles.name, ...this.props.style }} numberOfLines={1} >
          {this.props.location.name}
        </Text>
        <LocationItemDetail location={this.props.location} />
      </View>
    )
  }
}

LocationItem.propTypes = {
  location: PropTypes.object,
  style: PropTypes.any
}

const styles = StyleSheet.create({
  name: {
    fontSize: Layout.window.hp(2.75),
    fontWeight: 'bold',
    flex: 1
  },
  listItem: {
    height: Layout.window.hp(15),
    paddingVertical: Layout.window.hp(1.5),
    paddingHorizontal: Layout.window.wp(4)
  }
})
