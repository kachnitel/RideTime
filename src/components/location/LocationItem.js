import React from 'react'
import PropTypes from 'prop-types'
import { inject, observer } from 'mobx-react/native'
import {
  StyleSheet,
  Text,
  View,
  ActivityIndicator
} from 'react-native'
import Layout from '../../../constants/Layout'
import LocationItemDetail from './LocationItemDetail'
import { Location } from '../../stores/LocationStore.mobx'
import TrailforksLink from '../TrailforksLink'

export default
@inject('LocationStore')
@observer
class LocationItem extends React.Component {
  location: Location
  state = {
    loading: true
  }

  async componentDidMount () {
    this.location = await this.props.LocationStore.getAsync(this.props.locationId)
    this.setState({ loading: false })
  }

  render () {
    return (
      this.state.loading
        ? <ActivityIndicator />
        : <View style={styles.container}>
          <View style={styles.headerContainer}>
            <Text style={styles.name} numberOfLines={1} >
              {this.location.name}
            </Text>
            <TrailforksLink relativeUrl={'region/' + this.location.alias + '/'} />
          </View>
          <LocationItemDetail location={this.location} />
        </View>
    )
  }
}

LocationItem.propTypes = {
  locationId: PropTypes.number,
  style: PropTypes.any
}

const styles = StyleSheet.create({
  name: {
    fontSize: Layout.window.hp(2.75),
    flex: 1
  },
  headerContainer: {
    flexDirection: 'row'
  }
})
