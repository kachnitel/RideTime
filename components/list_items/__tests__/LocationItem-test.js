/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'
import LocationItem from '../LocationItem'
import locationStore from '../../../stores/LocationStore.singleton'
import { Location } from '../../../stores/LocationStore.mobx'

jest.mock('../../location/LocationDifficulties', () => 'LocationDifficulties')
jest.mock('../../RiderCount', () => 'RiderCount')

const mockMath = Object.create(global.Math)
mockMath.random = () => 0.5
global.Math = mockMath

it('renders correctly', async () => {
  let testLocation = {
    id: 1,
    name: 'Place',
    difficulties: [1, 2]
  }

  let location = new Location(locationStore)
  location.populateFromApiResponse(testLocation)
  locationStore.add(location)

  const component = renderer.create(
    <LocationItem.wrappedComponent
      locationId={location.id}
      LocationStore={locationStore}
    />
  )
  expect(component.toJSON()).toMatchSnapshot()
  let instance = component.getInstance()
  await new Promise((resolve) => setTimeout(resolve, 10))
  expect(await instance.location).toBe(location)
  expect(component.toJSON()).toMatchSnapshot()
})
