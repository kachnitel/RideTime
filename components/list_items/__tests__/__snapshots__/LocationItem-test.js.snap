// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ActivityIndicator
  animating={true}
  color="#999999"
  hidesWhenStopped={true}
  size="small"
  style={
    Object {
      "height": 200,
      "paddingHorizontal": 30,
      "paddingVertical": 20,
    }
  }
/>
`;

exports[`renders correctly 2`] = `
<View
  style={
    Object {
      "height": 200,
      "paddingHorizontal": 30,
      "paddingVertical": 20,
    }
  }
>
  <Text
    numberOfLines={1}
    style={
      Object {
        "flex": 1,
        "fontSize": 36.5,
      }
    }
  >
    Place
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <LocationDifficulties
      difficulties={
        Array [
          1,
          2,
        ]
      }
      iconSize={53.5}
      style={
        Object {
          "backgroundColor": "rgba(220, 220, 220, .5);",
          "borderRadius": 11.5,
          "flexDirection": "row",
          "padding": 7.5,
        }
      }
    />
    <RiderCount
      count={6}
      fontStyle={
        Object {
          "fontSize": 40,
        }
      }
      size={53.5}
    />
  </View>
</View>
`;
